### ISSUE TITLE
Support .m3u cleanup in `Import Content -> Scan Directory`

### ISSUE THREAD

**Issue by aktau** [libretro/RetroArch#10877]
## Description

Scanning for roms using the "automatic" Scan Directory functionality doesn't seem to do the m3u cleanup (announced as part of RetroArch 1.8.6). The only way I've been able to trigger this is using the "Manual Scan" option (I got this by reading the code).

The blog post talks about how the "Playlist Cleanup" action now also takes care of content that's in .m3u's too, but that didn't seem to work for me.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.8.8 (85d60bac8f)

### Environment information

- OS: Linux (Debian Bullseye), kernel 5.6
- Compiler: GCC 9

**Comment by jdgleaver** [comment]
Yes, M3U handling is only implemented for the manual scanner - I guess it can be added to the database scanner, but that code isn't much fun to deal with...

However, the `Clean Playlist` option should certainly handle this - if it doesn't, then it's a bug I've not encountered before. Could you please post just your playlist + M3U files (not the actual disk images, of course!) so I can investigate?

**Comment by aktau** [comment]
> However, the Clean Playlist option should certainly handle this - if it doesn't, then it's a bug I've not encountered before.

I'll try it again tonight, because yesterday I "fixed" the playlist situation by deleting the automatically generated one and replacing with a manual scan.

> Yes, M3U handling is only implemented for the manual scanner - I guess it can be added to the database scanner, but that code isn't much fun to deal with...

Sad to hear, I would've suspected that the database scanner "just" sets up the parameters that one manually has to feed into `Scan directory` (manual scan) so that improvements could be re-used. But I did not look further into the code than what I needed to confirm that the version I compiled contained (a) m3u cleanup code and (b) this was located int he manual scan code.

**Comment by jdgleaver** [comment]
> I would've suspected that the database scanner "just" sets up the parameters that one manually has to feed into Scan directory (manual scan) so that improvements could be re-used.

The database scanner is old code, and quite a different beast...

That's one of the reasons I implemented the manual scanner - I wanted something simple, clean and extensible, that wouldn't cause headaches. I can probably plug some M3U clean-up code into the database scan task, but I can't give you an ETA - we're all quite busy at the moment (what with Steam preparations, etc.)

**Comment by WilsonBradley** [comment]
REQUEST: Please fork/port to Oculus Go/ Quest. Oculus Arcade was dropped, to bad they didn't just make it open source. It's a perfect example.. However, ROMs can't be brought into it.

**Comment by twinaphex** [comment]
@WilsonBradley Please make this a separate Github issue. Going to remove it here as it detracts from the conversation going on here. Please don't post here again and instead make a separate issue.
